packages/dapp/src/components/TopNavigation.tsx

```diff
@@ -0,0 +1,41 @@
+import {useNavigate} from "react-router-dom";
+import {useAppState} from '../store';
+import {Avatar, Button, Header, Nav} from 'grommet';
+
+import {ThemeMode} from '../hooks/useStyle';
+import {CurrentPage} from '../hooks/usePageNav';
+import {Switch} from './Switch';
+
+const logoLink = 'wt-logo.png';
+
+export const TopNavigation = () => {
+  const navigate = useNavigate();
+
+  const { themeMode, switchThemeMode } = useAppState();
+  const { currentPage, switchCurrentPage } = useAppState();
```

Comment:
it is better to call hooks inside a component only once. So, these two calls of `useAppState` can be concatenated into one:

```
const { themeMode, switchThemeMode, currentPage, switchCurrentPage } = useAppState();
```